token/js/test/e2e-2022/tokenMetadata.test.ts

```diff
@@ -0,0 +1,181 @@
+        ])
+    );
+
+    before(async () => {
+        connection = await getConnection();
+        payer = await newAccountWithLamports(connection, 1000000000);
+    });
+
+    beforeEach(async () => {
+        mint = Keypair.generate();
+
+        // TODO @mistersimon:- Change to instruction creator once merged
+        const metadataPointerInstruction = new TransactionInstruction({
+            keys: [{ pubkey: mint.publicKey, isSigner: false, isWritable: true }],
+            programId: TEST_PROGRAM_ID,
+            data: Buffer.concat([new Uint8Array([39, 0]), authority.publicKey.toBuffer(), mint.publicKey.toBuffer()]),
+        });
+
+        const lamports = await connection.getMinimumBalanceForRentExemption(mintLen);
+
+        const transaction = new Transaction().add(
+            SystemProgram.createAccount({
+                fromPubkey: payer.publicKey,
+                newAccountPubkey: mint.publicKey,
+                space: mintLen,
+                lamports: lamports * 10,
+                programId: TEST_PROGRAM_ID,
+            }),
+            metadataPointerInstruction,
+            createInitializeMintInstruction(
+                mint.publicKey,
+                TEST_TOKEN_DECIMALS,
+                authority.publicKey,
+                null,
+                TEST_PROGRAM_ID
+            )
+        );
+
+        await sendAndConfirmTransaction(connection, transaction, [payer, mint], undefined);
+
+        await initializeTokenMetadata(
+            connection,
+            payer,
+            authority.publicKey,
+            mint.publicKey,
+            authority,
+            'name',
+            'symbol',
+            'uri',
+            undefined,
+            undefined,
+            TEST_PROGRAM_ID
+        );
+    });
+
+    const getTokenMetadata = async () => {
+        const mintInfo = await getMint(connection, mint.publicKey, undefined, TOKEN_2022_PROGRAM_ID);
+        const data = getExtensionData(19 as ExtensionType, mintInfo.tlvData);
+        if (data === null) {
+            throw new Error('No Extension Data Found');
+        }
+
+        const tlv = Buffer.concat([TOKEN_METADATA_DISCRIMINATOR, lengthBuffer(data), data]);
+        const meta = unpack(tlv);
+
+        return meta;
+    };
+
+    it('can successfully initialize', async () => {
+        const meta = await getTokenMetadata();
+        expect(meta).to.deep.equal({
+            updateAuthority: new PublicKey('ExgT3gCWXJzY4a9SHqTqsTk6dPAj37WNq2uWNbmMG1JR'),
+            mint: mint.publicKey,
+            name: 'name',
+            symbol: 'symbol',
+            uri: 'uri',
+            additionalMetadata: [],
+        });
+    });
+
+    it('can successfully emit', async () => {
+        const signature = await emitTokenMetadata(connection, payer, mint.publicKey);
+        const tx: any = await connection.getTransaction(signature);
+        const data = Buffer.from(tx?.meta?.returnData?.data?.[0], 'base64');
```

Comment:
Is the correct approach? The typescript doesn't have `returnData` defined on `tx.meta`, but its outputted when i run the test.